smart-contracts/contracts/mixins/MixinRefunds.sol

```diff
@@ -39,19 +44,40 @@ contract MixinRefunds is
   function cancelAndRefund()
     external
   {
-    Key storage key = _getKeyFor(msg.sender);
+    _cancelAndRefund(msg.sender);
+  }
 
-    uint refund = _getCancelAndRefundValue(msg.sender);
+  /**
+   * @dev Cancels a key owned by a different user and sends the funds to the msg.sender.
+   * @param _keyOwner this user's key will be canceled
+   * @param _signature getCancelAndRefundApprovalHash signed by the _keyOwner
+   */
+  function cancelAndRefundFor(
+    address _keyOwner,
+    bytes calldata _signature
+  ) external
+  {
+    require(
```

Comment:
So this proves the signer is the `_keyOwner`.

Reply:
Yes.  It verifies keyOwner signed it, and what they signed specifically included the 3 points from the getHash function.